app/controllers/api/v1/notes_controller.rb

```diff
@@ -0,0 +1,57 @@
+module Api
+  module V1
+    class NotesController < ApplicationController
+      rescue_from ActiveRecord::RecordNotFound, with: :render_record_not_found
+
+      def index
+        render json: notes, status: :ok, each_serializer: IndexNoteSerializer
+      end
```

Comment:
solemos validar los parámetros antes, por ejemplo:
```suggestion
      def index
       return invalid_type_error unless valid_note_type?
        render json: notes, status: :ok, each_serializer: IndexNoteSerializer
      end
```

Luego, por ejemplo:

```ruby
def invalid_type_error
  render json: "Algun mensaje de error", status: :algun_codigo_de_error
end
```

Por otro lado, el type es opcional, entonces si no está presente no quisieramos validar nada. Podriamos agregar algo mas a tu método que sea:

```ruby

 def valid_note_type?
        note_type.nil? || Note.note_types.keys.include?(note_type)
      end

def note_type
  params.permit[:note_type]
end
```

Reply:
Listo Flor! cree estos métodos para controlar el error del note_type en las query params: 
![image](https://github.com/ross-garcia/Training-Rails-UGUITO-API-Bootstrap/assets/160248891/0877328e-4317-48c3-9bc8-8c00690b9050)

Modifiqué validate_note_type? pero el método note_type rompía la request cuando no se mandaba el parámetro note_type. Por esta razón lo dejé de esta manera: 
![image](https://github.com/ross-garcia/Training-Rails-UGUITO-API-Bootstrap/assets/160248891/324e871c-4d2f-40b2-8c80-b64f4c8f7f48)

También pasé el mensaje de error de ActiveRecord::RecordNotFound que utilicé para show a application_controller  ya que puedo reutilizarlo con un mensaje genérico como me explicó Juaco